src/core/processing/qgsprocessingcontext.h

```diff
@@ -680,6 +681,30 @@ class CORE_EXPORT QgsProcessingContext
      */
     void setTemporaryFolder( const QString &folder );
 
+    /**
+     * Returns the (optional) number of threads to use when running algorithms.
+     *
+     * If set, this overrides the standard global Processing number of threads setting.
+     * Note that if algorithm implementation does not support multhreaded execution, this
+     * setting will be ignored.
+     *
+     * \see setNumberOfThreads()
+     * \since QGIS 3.32
+     */
+    int numberOfThreads() const;
```

Comment:
maybe this should be "maximumThreads" instead? It's more reflective of the actual use of the setting.